src/Microsoft.AspNetCore.Mvc.Abstractions/ModelBinding/Metadata/IModelBindingMessageProvider.cs

```diff
@@ -25,6 +25,13 @@ public interface IModelBindingMessageProvider
         Func<string> MissingKeyOrValueAccessor { get; }
 
         /// <summary>
+        /// Error message the model binding system adds when no value is provided for the request body,
+        /// but a value is required.
+        /// </summary>
+        /// <value>Default <see cref="string"/> is "A value for the request body was not provided.".</value>
+        Func<string> MissingRequestBodyRequiredValueAccessor { get; }
```

Comment:
@dougbu What's the process for adding a new validation error message, if we're keen not to have a breaking change on this? Are we going to do `IModelBindingMessageProvider2` and all that?

> That message (by default A value for the '{0}' property was not provided.) would be correct but doesn't mention the request body.

*Sort of* correct, but not totally, because it's not a "property" that we're lacking a value for. It's either a method parameter or a request body.

I'm happy to use that message if we think it's good enough, though it does seem a bit odd.